libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -987,10 +996,37 @@ struct SyncState {
 
 #[tonic::async_trait]
 impl NodeAPI for Greenlight {
-    fn node_credentials(&self) -> NodeResult<Option<NodeCredentials>> {
+    async fn reconnect(&self, shutdown: mpsc::Receiver<()>) {
+        debug!("Reconnect hibernation: request received");
+
+        // Force refresh existing grpc clients
+        *self.gl_client.lock().await = None;
+        *self.node_client.lock().await = None;
+
+        // Create a new signer
+        debug!("Reconnect hibernation: creating new signer");
+        let new_signer = match Signer::new(
+            self.seed.clone(),
+            self.sdk_config.network.into(),
+            self.device.clone(),
+        ) {
+            Ok(new_signer) => new_signer,
+            Err(e) => {
+                error!(
+                    "Reconnect hibernation: failed to create new signer after reconnect request: {:?}",
+                    e
+                );
+                return;
```

Comment:
It seems at this point if there's an error the previous signer loop is shutdown and we don't attempt to reconnect again (maybe it will if the hibernation reconnect is triggered again)? Is there a nice way to handle this?

Reply:
I pushed some changes after review comments from @roeierez. Now the `reconnect` function no longer starts the signer. Starting the signer is done in a loop, if starting the signer exits, it will try to start again after a sleep. It still won't recreate the signer, but it could still be recreated after a new hibernation.

I'm not sure if there's a better way to handle that. Retrying to create the signer I don't think is an option. Keeping the old signer is probably the best option if recreating it fails.